### PR TITLE
Lock issues closed for 3 months and pull requests merged/closed for 3 months automatically

### LINKED ISSUE #96: Lock issues closed for 3 months and pull requests merged/closed for 3 months automatically
To avoid noise from the closed/merged issues/pull requests to ask users to create new issues, I think it's necessary to lock all old issues/pull requests.

### PR DESCRIPTION
**Pull request by yardenshoham**
We will lock issues and pull requests that have been closed for 3 months. If the issue was updated in the last month, a comment will be posted suggesting opening a new issue to continue the discussion.

The check is done on every push and will lock at most 30 issues at a time. We currently have ~400 issues awaiting for locking so the initial locking will happen over the next few days.

Closes #96

### CODE REVIEW

**Comment by yardenshoham** [comment]
I don't know if it works as I don't have an admin token for go-gitea/gitea so I couldn't test it.

File: src/lock.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      ) => {
+        const lockedSuccessfully = await lockIssue(issue.number, "resolved");
+
+        // if the issue was updated in the last month, we add a comment
+        if (lockedSuccessfully && new Date(issue.updated_at) > oneMonthAgo) {
+          await addComment(
+            issue.number,
+            `Hi, we lock ${
+              issue.pull_request ? "pull request" : "issue"
+            }s if 3 months have passed since they were closed. If there's any need for a further discussion, please open a new issue. :tea:`,
```
Comment (delvh) [inline src/lock.ts:27]:
```suggestion
            }s 3 months after they were closed. If there's any need for further discussion, please open a new issue. :tea:`,
```

File: src/lock.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          updated_at: string;
+        },
+      ) => {
+        const lockedSuccessfully = await lockIssue(issue.number, "resolved");
+
+        // if the issue was updated in the last month, we add a comment
+        if (lockedSuccessfully && new Date(issue.updated_at) > oneMonthAgo) {
+          await addComment(
+            issue.number,
+            `Hi, we lock ${
```
Comment (delvh) [inline src/lock.ts:25]: That `Hi` completely throws me off-guard…
```suggestion
            `We lock ${
```

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
No comment at all would still be my preference, or a way to ignore these notifications, but GitHub does not offer a way unfortunately.

**Pull request merged**